### ISSUE TITLE
When the resource model is enabled, the karmada-controller-manager panic

### ISSUE THREAD

**Issue by Fish-pro** [karmada-io/karmada#3601]
**What happened**:

The following error occurs after the resource model is enabled
<img width="1099" alt="image" src="https://github.com/karmada-io/karmada/assets/47143494/8791f17d-cea0-4e1f-857c-db21b8403a51">

![企业微信截图_16854127816585](https://github.com/karmada-io/karmada/assets/47143494/cda82a95-6553-4aac-8c88-44ae3e781010)

![企业微信截图_16854138557404](https://github.com/karmada-io/karmada/assets/47143494/74cb760b-1281-4831-9b90-58519cdfe7f8)

I try to locate it，`searchLastLessElement` return '-1' 
```go
func searchLastLessElement(nums []resource.Quantity, target resource.Quantity) int {
	low, high := 0, len(nums)-1
	for low <= high {
		mid := low + ((high - low) >> 1)
		diff1 := nums[mid].Cmp(target)
		var diff2 int
		if mid != len(nums)-1 {
			diff2 = nums[mid+1].Cmp(target)
		}
		// diff < 1 means nums[mid] <= target
		// diff == 1 means nums[mid+1] > target
		if diff1 < 1 {
			if (mid == len(nums)-1) || (diff2 == 1) {
				// find the last less element that equal to target element
				return mid
			}
			low = mid + 1
		} else {
			high = mid - 1
		}
	}
	return -1
}
```
Get the slice object according to index '-1', thus panic
```go

// AddToResourceSummary add resource node into modeling summary
func (rs *ResourceSummary) AddToResourceSummary(crn ClusterResourceNode) {
	index := rs.getIndex(crn)
	modeling := &(*rs)[index]
	if rs.GetNodeNumFromModel(modeling) <= 5 {
		root := modeling.linkedlist
		if root == nil {
			root = list.New()
		}
		found := false
		// traverse linkedlist to add quantity of recourse modeling
		for element := root.Front(); element != nil; element = element.Next() {
			switch clusterResourceNodeComparator(element.Value, crn) {
			case 0:
				{
					tmpCrn := element.Value.(ClusterResourceNode)
					safeChangeNum(&tmpCrn.quantity, crn.quantity)
					element.Value = tmpCrn
					found = true
					break
				}
			case 1:
				{
					root.InsertBefore(crn, element)
					found = true
					break
				}
			case -1:
				{
					continue
				}
			}
			if found {
				break
			}
		}
		if !found {
			root.PushBack(crn)
		}
		modeling.linkedlist = root
	} else {
		root := modeling.redblackTree
		if root == nil {
			root = llConvertToRbt(modeling.linkedlist)
			modeling.linkedlist = nil
		}
		tmpNode := root.GetNode(crn)
		if tmpNode != nil {
			node := tmpNode.Key.(ClusterResourceNode)
			safeChangeNum(&node.quantity, crn.quantity)
			tmpNode.Key = node
		} else {
			root.Put(crn, crn.quantity)
		}
		modeling.redblackTree = root
	}
	safeChangeNum(&modeling.Quantity, crn.quantity)
}
```

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Karmada version: latest
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by RainbowMango** [comment]
Which version are you using? We just fixed a panic issue at #3591 yesterday.

**Comment by Fish-pro** [comment]
> Which version are you using? We just fixed a panic issue at #3591 yesterday.

@RainbowMango 
I installed it last week. we did not update the local branch. I will try again after updating

**Comment by RainbowMango** [comment]
Why closed this?
